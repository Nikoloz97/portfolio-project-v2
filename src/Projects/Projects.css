.project-card {
  color: white;
  align-items: center;
  padding: 0.5rem !important;
  box-shadow: 0 0 10px rgba(255, 255, 255, 0.7) !important;
  margin-top: 2rem !important;
  background-color: #383c43 !important;
  position: relative;

  transition: height 1s cubic-bezier(0.5, 0, 0.2, 1),
    transform 1s cubic-bezier(0.4, 0, 0.2, 1), opacity 1s ease-in !important;

  width: 20.7rem !important;
}

/* On render animation */
.project-card-preloaded {
  height: 5em !important;
  transform: scaleY(0) !important;
  transform-origin: center !important;
  opacity: 0.1 !important;
}

.project-card-loaded {
  height: 20em !important;
  transform: scaleY(1) !important;
  opacity: 1 !important;
}

/* Left-right arrow button positions */

.project-left-arrow-button {
  position: absolute;
  width: 20px;
  top: 0;
  left: 0;
  bottom: 0;
}

.project-right-arrow-button {
  position: absolute;
  width: 20px;
  top: 0;
  left: 0;
  bottom: 0;
  transform: translateX(591%);
}
