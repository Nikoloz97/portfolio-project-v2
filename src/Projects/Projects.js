import React from "react";
import { Outlet, Router } from "react-router-dom";

const Projects = () => {
  return (
    <div className="Default-Page">
      <header>
        <p>Welcome to my projects page</p>
      </header>
      <div>
        <Outlet />
      </div>
    </div>
  );
};

export default Projects;
