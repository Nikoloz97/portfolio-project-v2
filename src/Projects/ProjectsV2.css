.Projects2-Page {
  background: url("../Images/Projects/notebook-near-measuring-tools.jpg");
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100vh;
  width: 100vw;
  filter: brightness(0.8);
  color: white;
}

.Projects-Page::before {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  background: linear-gradient(
      to top,
      rgb(13, 20, 21) 0%,
      rgba(5, 14, 19, 0) 20%
    ),
    linear-gradient(
      to bottom,
      rgba(13, 20, 21, 0.473) 8%,
      rgba(5, 14, 19, 0) 40%
    );
  width: 100vw;
  height: 100%;
}

.Projects2-Container {
  display: flex;
  width: 100vw;
  height: 80vh;
  justify-content: center;
  align-items: center;
  gap: 10%;
}

.Projects2-Title-Description-Container {
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: start;
  align-items: center;
  width: 40%;
}

.Projects2-Title {
  font-size: 3em;
  margin-top: 1em;
}

.Projects2-Project-Title {
  font-size: 5em;
  font-weight: 400;
  margin-top: 3em;
}

.Projects2-Project-Description {
  margin-top: 4em;
}

.Projects2-Project-Button {
  background-color: rgba(128, 128, 128, 0.2) !important;
  margin-top: 2em !important;
  font-size: 1.5em !important;
}

.Projects2-Images-Container {
  display: flex;
  width: 60%;
  gap: 1em;
}

.Projects2-Image {
  max-height: 200px;
}

.Projects2-Image:hover {
  cursor: pointer;
}
