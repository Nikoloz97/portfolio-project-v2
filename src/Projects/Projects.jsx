import React, { useState, useEffect } from "react";
import { Card, Image, Button, Icon } from "semantic-ui-react";
import { Link } from "react-router-dom";
import "./Projects.css";

const Projects = () => {
  const [currentProjectIndex, setCurrentProjectIndex] = useState(0);
  const [isLoaded, setIsLoaded] = useState(false);
  const [buttonIndexForInversion, setButtonIndexForInversion] = useState(null);

  const projectsData = [
    {
      introText: "Projects",
      introSubtext:
        "Select the arrow on the right to navigate to my various projects - or click on the links below",
      linkUrls: [
        { title: "Kronos", url: "/projects/kronos" },
        { title: "Calculator", url: "/projects/calculator" },
        { title: "Geography Game", url: "/projects/geography-game" },
        { title: "Fantasy Basketball", url: "/projects/fantasy-basketball" },
      ],
    },
    {
      title: "Kronos",
      mediaUrl: "../../Images/Projects/Kronos2.jpg",
      mediaCaption: "This is a description about the clocks project",
      linkUrl: "/projects/kronos",
    },
    {
      title: "Calculator",
      mediaUrl: "../../Images/Projects/Calc1.jpg",
      mediaCaption: "This is a description about the calculator project",
      linkUrl: "/projects/calculator",
    },
    {
      title: "Geography Game",
      mediaUrl: "../../Images/Projects/Globe1.jpg",
      mediaCaption: "This is a description about the geography game project",
      linkUrl: "/projects/geography-game",
    },
    {
      title: "Fantasy Basketball",
      mediaUrl: "../../Images/Projects/Nba1.jpg",
      mediaCaption:
        "This is a description about the fantasy basketball project",
      linkUrl: "/projects/fantasy-basketball",
    },
  ];

  useEffect(() => {
    const timer = setTimeout(() => {
      setIsLoaded(true);
    }, 200);

    return () => {
      clearTimeout(timer);
    };
  });

  const handleNextProject = () => {
    if (currentProjectIndex < projectsData.length - 1) {
      setIsLoaded(false);

      setTimeout(() => {
        setCurrentProjectIndex(currentProjectIndex + 1);
        setIsLoaded(true);
      }, 200);
    }
  };

  const handlePrevProject = () => {
    if (currentProjectIndex <= projectsData.length - 1) {
      setIsLoaded(false);

      setTimeout(() => {
        setCurrentProjectIndex(currentProjectIndex - 1);
        setIsLoaded(true);
      }, 200);
    }
  };

  return (
    <div className="Default-Page">
      {/* Intro card */}
      {currentProjectIndex === 0 ? (
        <Card
          className={`${
            isLoaded
              ? "project-card project-card-loaded"
              : "project-card-preloaded"
          }`}
        >
          <Card.Header
            style={{ maxWidth: "70%", marginBottom: "10%", marginTop: "10%" }}
            textAlign="center"
          >
            {projectsData[currentProjectIndex].introText}
          </Card.Header>
          <Card.Description
            style={{
              maxWidth: "70%",
              fontSize: "1.2rem",
              opacity: "0.7",
            }}
            textAlign="center"
          >
            {projectsData[currentProjectIndex].introSubtext}
          </Card.Description>
          <Card.Content extra style={{ maxWidth: "70%" }} textAlign="center">
            <Button
              aria-label="Previous project"
              className="project-left-arrow-button"
              disabled={true}
            >
              <Icon name="arrow left" style={{ marginLeft: "-8px" }} />
            </Button>

            {projectsData[currentProjectIndex].linkUrls.map((link, index) => (
              <div
                onMouseEnter={() => setButtonIndexForInversion(index)}
                onMouseLeave={() => setButtonIndexForInversion(null)}
                key={index}
              >
                <Button
                  as={Link}
                  to={link.url}
                  style={{ marginBottom: "10%" }}
                  inverted={buttonIndexForInversion === index ? false : true}
                >
                  {link.title}
                </Button>
              </div>
            ))}

            <Button
              aria-label="Next project"
              className="project-right-arrow-button"
              disabled={
                currentProjectIndex === projectsData.length - 1 ? true : false
              }
              onClick={handleNextProject}
            >
              <Icon name="arrow right" style={{ marginLeft: "-7px" }} />
            </Button>
          </Card.Content>
        </Card>
      ) : (
        /* Project card */
        <Card
          className={`${
            isLoaded
              ? "project-card project-card-loaded"
              : "project-card-preloaded"
          }`}
        >
          <Card.Header
            style={{ maxWidth: "70%", marginBottom: "10%", marginTop: "10%" }}
            textAlign="center"
          >
            {projectsData[currentProjectIndex].title}
          </Card.Header>
          <Image
            style={{ maxWidth: "70%", marginBottom: "5%" }}
            src={require("../Images/Projects/Calc1.jpg")}
            wrapped
            ui={false}
          />
          <Card.Description
            style={{ maxWidth: "70%", fontSize: "1.2rem", opacity: "0.7" }}
            textAlign="center"
          >
            {projectsData[currentProjectIndex].mediaCaption}
          </Card.Description>
          <Card.Content
            extra
            style={{
              display: "flex",
              justifyContent: "space-between",
              alignItems: "center",
              maxWidth: "70%",
            }}
            textAlign="center"
          >
            <Button
              aria-label="Previous project"
              className="project-left-arrow-button"
              onClick={handlePrevProject}
            >
              <Icon name="arrow left" style={{ marginLeft: "-8px" }} />
            </Button>
            <Button
              as={Link}
              to={projectsData[currentProjectIndex].linkUrl}
              inverted
            >
              Go to Project
            </Button>
            <Button
              aria-label="Next project"
              className="project-right-arrow-button"
              disabled={
                currentProjectIndex === projectsData.length - 1 ? true : false
              }
              onClick={handleNextProject}
            >
              <Icon name="arrow right" style={{ marginLeft: "-7px" }} />
            </Button>
          </Card.Content>
        </Card>
      )}
    </div>
  );
};

export default Projects;
