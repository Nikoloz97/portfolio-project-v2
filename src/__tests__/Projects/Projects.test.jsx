import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import Projects from "../../Projects/Projects";
import { MemoryRouter } from "react-router-dom";

describe("Projects", () => {
  it("produces intro card where prev/next buttons are disabled/enabled, respectively", () => {
    render(
      <MemoryRouter>
        <Projects />
      </MemoryRouter>
    );

    expect(screen.getByText("Projects")).toBeInTheDocument();
    expect(
      screen.getByRole("button", { name: "Previous project" })
    ).toHaveAttribute("disabled");
    expect(
      screen.getByRole("button", {
        name: "Next project",
      })
    ).not.toHaveAttribute("disabled");
  });

  it("selecting right arrow button goes to next card and right arrow button is not disabled", async () => {
    render(
      <MemoryRouter>
        <Projects />
      </MemoryRouter>
    );

    const rightIntroArrowButton = screen.getByRole("button", {
      name: "Next project",
    });

    userEvent.click(rightIntroArrowButton);

    await waitFor(() => {
      const rightProjectArrowButton = screen.getByRole("button", {
        name: "Next project",
      });

      expect(screen.getByText("Kronos")).toBeInTheDocument();
      expect(rightProjectArrowButton).not.toHaveAttribute("disabled");
    });
  });
});
