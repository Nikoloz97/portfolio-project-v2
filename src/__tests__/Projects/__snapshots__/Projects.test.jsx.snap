// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects matches snapshot following changes 1`] = `
<div
  className="Default-Page"
>
  <div>
    <div
      className="ui card project-card-preloaded"
      onClick={[Function]}
    >
      <div
        className="center aligned header"
        style={
          Object {
            "marginBottom": "10%",
            "marginTop": "10%",
            "maxWidth": "70%",
          }
        }
      >
        Projects
      </div>
      <div
        className="center aligned description"
        style={
          Object {
            "fontSize": "1.2rem",
            "maxWidth": "70%",
            "opacity": "0.7",
          }
        }
      >
        Select the arrow on the right to navigate to my various projects - or click on the links below
      </div>
      <div
        className="extra center aligned content"
        style={
          Object {
            "maxWidth": "70%",
          }
        }
      >
        <button
          aria-label="Previous project"
          className="ui disabled button project-left-arrow-button"
          disabled={true}
          onClick={[Function]}
          tabIndex={-1}
        >
          <i
            aria-hidden="true"
            className="arrow left icon"
            onClick={[Function]}
            style={
              Object {
                "marginLeft": "-8px",
              }
            }
          />
        </button>
        <div
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="ui inverted button"
            href="/projects/kronos"
            onClick={[Function]}
            role="button"
            style={
              Object {
                "marginBottom": "10%",
              }
            }
          >
            Kronos
          </a>
        </div>
        <div
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="ui inverted button"
            href="/projects/calculator"
            onClick={[Function]}
            role="button"
            style={
              Object {
                "marginBottom": "10%",
              }
            }
          >
            Calculator
          </a>
        </div>
        <div
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="ui inverted button"
            href="/projects/geography-game"
            onClick={[Function]}
            role="button"
            style={
              Object {
                "marginBottom": "10%",
              }
            }
          >
            Geography Game
          </a>
        </div>
        <div
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="ui inverted button"
            href="/projects/fantasy-basketball"
            onClick={[Function]}
            role="button"
            style={
              Object {
                "marginBottom": "10%",
              }
            }
          >
            Fantasy Basketball
          </a>
        </div>
        <button
          aria-label="Next project"
          className="ui button project-right-arrow-button"
          onClick={[Function]}
        >
          <i
            aria-hidden="true"
            className="arrow right icon"
            onClick={[Function]}
            style={
              Object {
                "marginLeft": "-7px",
              }
            }
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
